consensus/replay.go

```diff
@@ -259,12 +259,9 @@ func (h *Handshaker) Handshake(proxyApp proxy.AppConns) error {
 		"protocol-version", res.AppVersion,
 	)
 
-	// Set AppVersion on the state.
-	if h.initialState.Version.Consensus.App != version.Protocol(res.AppVersion) {
+	if h.initialState.LastBlockHeight == 0 { //we only update state when we are in initial state
```

Comment:
why we only update state if LastBlockHeight == 0? say version changes at block 10. Don't we want to update the state in that case?